flask_calendar/actions.py

```diff
@@ -14,16 +13,18 @@
                                       get_session_username, authorized)
 
 
-authentication = Authentication(
-    data_folder=config.USERS_DATA_FOLDER,
-    password_salt=config.PASSWORD_SALT,
-    failed_login_delay_base=config.FAILED_LOGIN_DELAY_BASE
-)
+def _create_authentication() -> Authentication:
+    return Authentication(
+        data_folder=current_app.config['USERS_DATA_FOLDER'],
+        password_salt=current_app.config['PASSWORD_SALT'],
+        failed_login_delay_base=current_app.config['FAILED_LOGIN_DELAY_BASE'],
+    )
 
 
 @authenticated
 def index_action() -> Response:
     username = get_session_username(session_id=str(request.cookies.get(SESSION_ID)))
+    authentication = _create_authentication()
```

Comment:
now the app is tiny, but if we increase the amount of actions that interact with authentication stuff, doing a per-action instantiation auth is worse than having a module-level instance. In python is perfectly fine to keep the module-level `authentication = Authentication(...)` sentence.

If as I fear this change is to avoid cyclic imports between `app.py` and `actions.py`, then let's leave it as it is and create me a GH Issue to see how to solve it (maybe Flask app object has some kind of registry, or else we can always treat it as a singleton with an accessor that creates it only the first time and stores it at module-level).

Reply:
There is the [flask.g](https://flask.palletsprojects.com/en/1.1.x/api/#flask.g) namespace object which is used to store per-request data. This object is used on examples [to handle SQLite3 database connection objects](https://flask.palletsprojects.com/en/1.1.x/patterns/sqlite3/), which seems like a similar use case to the one at hand.

It is more or less similar to the singleton you proposed, so I will refactor the code to use this object, which seems like a more Flask-y way of handling its creation.